redaxo/src/core/backend.php

```diff
@@ -222,13 +224,6 @@
 // page variable validated
 rex_extension::registerPoint(new rex_extension_point('PAGE_CHECKED', $page, ['pages' => $pages], true));
 
-// trigger api functions
-// If the backend session is timed out, rex_api_function would throw an exception
-// so only trigger api functions if page != login
-if ('login' != $page) {
-    rex_api_function::handleCall();
-}
```

Comment:
Zu der page-problematik gibt es auch ein Issue: https://github.com/redaxo/redaxo/issues/2030
Das wäre mit deiner Änderung dann erledigt. 👍 

Ich muss mir das aber nochmal genauer anschauen, ob ich irgendwelche Probleme sehe mit der neuen Position der Zeile.

Reply:
Zur Erklärung der Änderung: Verschiebung von rex_api_function::handleCall();

Annahme 1: rex::getUser() ist nur dann nicht null, wenn die Session aktiv ist.
Annahme 2: Der nachfolgende Code ist nur für das Anzeigen der Backend-Page und somit nicht für rex_api relevant.

Problematik bei der aktuellen Position ist, dass in rex_be_controller::checkPagePermissions($user); eine Umleitung auf die log-in Seite vorgenommen wird, wenn kein 'page' Parameter gesetzt ist.
Um die Kompatibilität mit PUT, DELETE, UPDATE, ... herzustellen, müsste dort auch die Request-Methode geprüft werden. 

Dies wäre bei der Verschiebung zu der vorgeschlagenen Stelle nicht notwendig.

Reply:
Nur um sicher zu sein, hier mein Verständnis .... rex_api ist dazu da, per URL angetriggerte Aktionen auszuführen, bevor die Seite aufgebaut wird. Früher mal war das ein Parameter "func=irgendwas", der dann in der page zuerst abgefragt wurde,  die irgendwas-Aktion ausführte und dann die Liste oder das Formular aufbaute. Da das schon mal zu ziemlich unübersichtlichen pages führte, was rex_api eine Variante, um den Aktions-Code auszulagern und einen schicken Aufrufmechanismus zu haben. rex_api für AJAX oder so einzusetzen, ist mißbräuchlich (oder netter gesagt: ein Hack).

Und in Sachen Login-Page statt Startseite: Warum soll ich als eingeloggter Backend-Benutzer im Fall der Fälle auf die Login-Seite geworfen werden? Insbesondere wenn ich der Entwickler bin und grade an einem API schraube. Und als nicht eingeloggter User müsste ich doch eh ohne Anmeldung auf der Login-Seite landen? 

Oder was habe ich mal wieder falsch verstanden?

Reply:
Ich tendiere dazu, der Änderung hier zuzustimmen.
Ich denke das passt so. So ganz sicher bin ich aber auch nicht.
Aber es ist ja das, was ihr auch in #2030 vorgeschlagen habt.

@staabm Was meinst du denn hierzu?